polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -3116,11 +3136,17 @@ ValueCategory MLIRScanner::VisitImag(UnaryOperator *E, QualType PromotionType) {
 ValueCategory MLIRScanner::VisitReal(UnaryOperator *E, QualType PromotionType) {
   Expr *Op = E->getSubExpr();
 
-  assert(!Op->getType()->isAnyComplexType() && "Unsupported");
+  assert(
+      (!Op->getType()->isAnyComplexType() || Op->getType()->isComplexType()) &&
+      "Unsupported");
```

Comment:
```
  bool isComplexType() const;      // C99 6.2.5p11 (complex)
  bool isAnyComplexType() const;   // C99 6.2.5p11 (complex) + Complex Int.
```
If I understand correctly, we want to assert that the type is not complex int.
```suggestion
      "Unsupported complex integer type");
```